packages/math/src/pow.rs

```diff
@@ -0,0 +1,202 @@
+
+    let mut term = Decimal256::one();
+    let mut sum = Decimal256::one();
+    let mut negative = false;
+
+    let mut i = 1u64;
+
+    loop {
+        if term < prec {
+            break;
+        }
+
+        let big_k = Decimal256::one() * Decimal256::from_ratio(i, 1u64);
+        let (c, c_neg) = abs_diff_with_sign(a, big_k - Decimal256::one());
+
+        term *= c * x;
+        term /= big_k;
+
+        if term.is_zero() {
+            break;
+        }
+
+        if x_neg {
+            negative = !negative;
+        }
+
+        if c_neg {
+            negative = !negative;
+        }
+
+        if negative {
+            sum -= term;
+        } else {
+            sum += term;
+        }
+
+        i += 1;
+    }
+
+    Ok(sum)
+}
+
+pub fn abs_diff_with_sign(x: Decimal256, y: Decimal256) -> (Decimal256, bool) {
+    (x.abs_diff(y), x > y)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use std::str::FromStr;
+
+    use cosmwasm_std::Decimal256;
+
+    #[test]
+    fn test_pow() -> anyhow::Result<()> {
+        struct Case {
+            base: Decimal256,
+            exp: Decimal256,
+            res: Decimal256,
+            err: Option<Decimal256>,
+        }
+
+        let cases = [
+            Case {
+                // medium base, small exp
+                base: Decimal256::from_str("0.8")?,
+                exp: Decimal256::from_str("0.32")?,
+                res: Decimal256::from_str("0.93108385")?,
+                err: Some(Decimal256::from_str("0.00000001")?),
+            },
+            Case {
+                // zero exp
+                base: Decimal256::from_str("0.8")?,
+                exp: Decimal256::zero(),
+                res: Decimal256::one(),
+                err: None,
+            },
+            Case {
+                // large base, small exp
+                base: Decimal256::from_str("1.9999")?,
+                exp: Decimal256::from_str("0.23")?,
+                res: Decimal256::from_str("1.172821461")?,
+                err: Some(Decimal256::from_str("0.00000001")?),
+            },
+            Case {
+                // small base, large exp
+                base: Decimal256::from_str("0.0000123")?,
+                exp: Decimal256::from_str("123")?,
+                res: Decimal256::zero(),
+                err: None,
+            },
+            Case {
+                // large base, large exp
+                base: Decimal256::from_str("1.777")?,
+                exp: Decimal256::from_str("20")?,
+                res: Decimal256::from_str("98570.862372081602")?,
+                err: Some(Decimal256::from_str("0.000000000001")?),
+            },
+            Case {
+                // base equal one
+                base: Decimal256::from_str("1")?,
+                exp: Decimal256::from_str("123")?,
+                res: Decimal256::one(),
+                err: None,
+            },
+        ];
+
+        for c in cases {
```

Comment:
no way to do a monkey test?
for example: compare results of our pow implementation and slow but accurate existing pow implementation.

Reply:
Currently, Due to osmomath is written in Golang, There's only way we can do it is call osmomath via FFI.

Reply:
it's hard to test with random generated ones, so i decided to use the same number as the osmomath tested

Reply:
ah okay got it :)